notebooks/demo_mazetokenizermodular.ipynb

```diff
@@ -0,0 +1,4851 @@
+      "counter1 - counter2 = Counter({'SOUTH': 5, '0': 4, 'EAST': 3})\n",
+      "  warn_func(msg, category=warn_cls, source=except_from)\n",
+      "c:\\Users\\mivan\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\maze-dataset-K69OfPxp-py3.11\\Lib\\site-packages\\muutils\\errormode.py:52: UserWarning: Rollouts are not the same length: 100 != 92\n",
+      "  warn_func(msg, category=warn_cls, source=except_from)\n",
+      "Tokenizers:  48%|████▊     | 4845/10000 [00:09<00:09, 533.81it/s]c:\\Users\\mivan\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\maze-dataset-K69OfPxp-py3.11\\Lib\\site-packages\\muutils\\errormode.py:52: UserWarning: Adjacency list counters are not the same:\n",
+      "Counter({'ADJ_GROUP': 4, '1': 4, ')': 4, '<XX>': 4, '0': 4, 'EAST': 3, 'SOUTH': 1})\n",
+      "!=\n",
+      "Counter({'ADJ_GROUP': 4, '1': 4, ')': 4, '<XX>': 4, 'WEST': 3, '2': 2, '0': 2, 'NORTH': 1})\n",
+      "counter1 - counter2 = Counter({'EAST': 3, '0': 2, 'SOUTH': 1})\n",
+      "  warn_func(msg, category=warn_cls, source=except_from)\n",
+      "c:\\Users\\mivan\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\maze-dataset-K69OfPxp-py3.11\\Lib\\site-packages\\muutils\\errormode.py:52: UserWarning: Adjacency list counters are not the same:\n",
+      "Counter({'(': 4, '0': 4, ',': 4, ')': 4, '<XX>': 4, '&': 4, ';': 4, '1': 4, 'EAST': 3, 'SOUTH': 1})\n",
+      "!=\n",
+      "Counter({'(': 4, ',': 4, '1': 4, ')': 4, '<XX>': 4, '&': 4, ';': 4, 'WEST': 3, '0': 2, '2': 2, 'NORTH': 1})\n",
+      "counter1 - counter2 = Counter({'EAST': 3, '0': 2, 'SOUTH': 1})\n",
+      "  warn_func(msg, category=warn_cls, source=except_from)\n",
+      "c:\\Users\\mivan\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\maze-dataset-K69OfPxp-py3.11\\Lib\\site-packages\\muutils\\errormode.py:52: UserWarning: Adjacency list counters are not the same:\n",
+      "Counter({'(': 8, '<-->': 8, '2': 8, '1': 6, 'NORTH': 5, 'WEST': 3, '0': 2})\n",
+      "!=\n",
+      "Counter({'(': 8, '<-->': 8, '0': 6, '1': 6, 'SOUTH': 5, '2': 4, 'EAST': 3})\n",
+      "counter1 - counter2 = Counter({'NORTH': 5, '2': 4, 'WEST': 3})\n",
+      "  warn_func(msg, category=warn_cls, source=except_from)\n",
+      "c:\\Users\\mivan\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\maze-dataset-K69OfPxp-py3.11\\Lib\\site-packages\\muutils\\errormode.py:52: UserWarning: Adjacency list counters are not the same:\n",
+      "Counter({'&': 12, 'ADJ_GROUP': 8, '(': 8, '<-->': 8, ';': 8, '0': 6, 'EAST': 6, 'SOUTH': 6, '1': 6, '<XX>': 4, '2': 4})\n",
+      "!=\n",
+      "Counter({'&': 12, 'ADJ_GROUP': 8, '(': 8, ';': 8, '<-->': 8, '1': 6, 'WEST': 6, '2': 6, 'NORTH': 6, '0': 4, '<XX>': 4})\n",
+      "counter1 - counter2 = Counter({'EAST': 6, 'SOUTH': 6, '0': 2})\n",
+      "  warn_func(msg, category=warn_cls, source=except_from)\n",
+      "Tokenizers:  49%|████▉     | 4908/10000 [00:09<00:09, 560.85it/s]c:\\Users\\mivan\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\maze-dataset-K69OfPxp-py3.11\\Lib\
```

Comment:
For some `TokenizerElement` selections, `equal_except_adj_list_sequence` is designed to generally return `False`. This is the case if using both `EdgeGroupings.ByLeadingCoord` and `EdgePermuters.RandomCoords`. In this case each edge is perrmuted randomly each time (leading coords are random). When grouping those edges together by leading coord, there is only 1 occurrence of each leading coord in the stringification. So different random leading coord selections will produce different token counters.
If this is acceptable behavior, I'll remove this check from the notebook. If not, let's talk about what's wrong.

Reply:
Put this test into all_tokenizers tests, do this check for all tokenizers only if those conditions are False

Reply:
recap of discussion during meeting:
- difference in counters is fine for these special cases
- still want to check counters for all other cases
- thus, put in a check for this special case in the for loop
- move out of notebook tests and into unit tests